data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/SqsQueueUrl.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class SqsQueueUrl {
+    private final URL queueUrl;
+    private final String accountId;
+
+    private SqsQueueUrl(final URL queueUrl) throws MalformedURLException {
+        this.queueUrl = queueUrl;
+        final String path = queueUrl.getPath();
+
+        if (path.isEmpty())
+            throw new MalformedURLException();
```

Comment:
`MalformedURLException`'s are "[t]hrown to indicate that a malformed URL has occurred. Either no legal protocol could be found in a specification string or the string could not be parsed." The failure scenarios in this class do not match this definition as they are related to the contents of the `path`. I see that you are trying to make this class return a single Exception but we should focus on the user experience and what would help with debugging. 

I would recommend returning an `IllegalArgumentException` and provide a descriptive reason for the exception so the user can easily debug this failure.

Reply:
Good callout. I was thinking of the "string could not be parsed" part as we cannot parse it as an SQS queue URL. But, that is probably intended to mean that it doesn't have the necessary components for parsing a URL. I'll change it.